numpyro/distributions/discrete.py

```diff
@@ -630,6 +630,9 @@ def mean(self):
     def variance(self):
         return self.rate
 
+    def cdf(self, value):
+        return 1 - gammainc(value + 1, self.rate)
```

Comment:
This implementation relies on value to be of type int. To be consistent with the definition of the cdf, and also with the scipy implementation, I'd use
 ```
   def cdf(self, value):
        k = jnp.floor(value) + 1
        return gammaincc(k, self.rate)
```

Reply:
Thanks, this is a good point!

(in the future, out-of-support values will raise a warning.)

Reply:
Well, even though the pmf only has integer support, the cdf has real values support. You should keep that in mind when adding the out-of-support warnings!

Reply:
Yup, but I think it is better to raise a warning and let users convert the value, rather than supporting a generic definition of cdf_support (though it makes sense to say that `Exponential.cdf(-1.) = 0`).